src/node.cc

```diff
@@ -2661,9 +2712,24 @@ void SetupProcessObject(Environment* env,
 
   process->SetAccessor(env->title_string(),
                        ProcessTitleGetter,
-                       ProcessTitleSetter,
+                       env->is_main_instance() ? ProcessTitleSetter : nullptr,
                        env->as_external());
 
+  // process.threadId
+  READONLY_PROPERTY(process,
+                    "threadId",
+                    Number::New(env->isolate(), env->thread_id()));
+
+  // process.isMainInstance
+  READONLY_PROPERTY(process,
+                    "isMainInstance",
+                    Boolean::New(env->isolate(), env->is_main_instance()));
+
+  // process.isWorkerInstance
+  READONLY_PROPERTY(process,
+                    "isWorkerInstance",
+                    Boolean::New(env->isolate(), env->is_worker_instance()));
```

Comment:
If the main thread is guaranteed to have a `process.threadId` of 0, there is no need to have `process.isMainInstance` nor `process.isWorkerInstance`. It also would help reduce the API surface.

On a related point, should it be `process.threadId` or `process.tid` as discussed in the last PR?